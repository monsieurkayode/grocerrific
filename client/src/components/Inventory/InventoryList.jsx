/* eslint-disable no-underscore-dangle */
import React from 'react';
import {
  func, arrayOf, shape, bool
} from 'prop-types';

import InventoryListItem from './InventoryListItem';
import Loader from '../common/Loader';
import NoContent from '../common/NoContent';

const InventoryList = ({
  openModal,
  openDeleteModal,
  groceries,
  loading
}) => (
  <div className="inventory__list">
    {loading && <Loader size={60} />}
    {!loading && groceries.length === 0 && (
    <NoContent
      content="There are currently no items in the Inventory"
    />
    )}
    {
      !loading && groceries.length > 0 && (
      <div className="table-header">
        <div className="name">Name</div>
        <div>Price</div>
        <div>Stock</div>
        <div className="action">&nbsp;</div>
      </div>
      )
    }
    {!loading && groceries.map(grocery => (
      <InventoryListItem
        key={grocery._id}
        openModal={openModal}
        openDeleteModal={openDeleteModal}
        {...grocery}
      />
    ))}
  </div>
);

InventoryList.propTypes = {
  openModal: func.isRequired,
  openDeleteModal: func.isRequired,
  groceries: arrayOf(shape({})).isRequired,
  loading: bool.isRequired
};

export default InventoryList;
