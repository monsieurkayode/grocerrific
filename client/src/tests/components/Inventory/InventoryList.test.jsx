import React from 'react';
import { shallow } from 'enzyme';
import InventoryList from '../../../components/Inventory/InventoryList';
import mockGroceries from '../../__mocks__/mockGroceries';

const setup = () => {
  const props = {
    groceries: mockGroceries,
    openModal: jest.fn(),
    openDeleteModal: jest.fn(),
    loading: false,
  };

  return shallow(<InventoryList {...props} />);
};

describe('InventoryList Component', () => {
  const wrapper = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render three InventoryListItem children components', () => {
    expect(wrapper.find('InventoryListItem').length).toBe(3);
  });

  it('should render a Loader component when loading', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper.find('Loader').length).toBe(1);
  });

  it('should render a NoContent component when there are no groceries', () => {
    wrapper.setProps({ groceries: [], loading: false });
    expect(wrapper.find('NoContent').length).toBe(1);
  });
});
