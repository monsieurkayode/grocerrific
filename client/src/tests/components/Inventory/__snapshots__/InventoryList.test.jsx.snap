// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryList Component renders correctly 1`] = `
<div
  className="inventory__list"
>
  <div
    className="table-header"
  >
    <div
      className="name"
    >
      Name
    </div>
    <div>
      Price
    </div>
    <div>
      Stock
    </div>
    <div
      className="action"
    >
      Â 
    </div>
  </div>
  <InventoryListItem
    _id="5bf4754223602e00211d6587"
    key="5bf4754223602e00211d6587"
    name="mamador"
    openDeleteModal={[MockFunction]}
    openModal={[MockFunction]}
    price={1250}
    quantity={3}
  />
  <InventoryListItem
    _id="5bf4754223602e00211d6588"
    key="5bf4754223602e00211d6588"
    name="sausage"
    openDeleteModal={[MockFunction]}
    openModal={[MockFunction]}
    price={120}
    quantity={12}
  />
  <InventoryListItem
    _id="5bf4754223602e00211d6589"
    key="5bf4754223602e00211d6589"
    name="salad creme"
    openDeleteModal={[MockFunction]}
    openModal={[MockFunction]}
    price={1500}
    quantity={0}
  />
</div>
`;
